### ISSUE TITLE
[distributed] Objectlayer fails to initialize properly under heavy load. 

### ISSUE THREAD

**Issue by harshavardhana** [minio/minio#2982]
Steps to reproduce run each of these commands in a while true loop using 'mc'  
- mc rm -r --force myminio/testbucket/
- mc ls -r myminio/testbucket/
- mc cp -r /usr myminio/testbucket

Keep running this and try to initialize server by freshly starting them. Server gets hung for the most part only serving object API requests and not proceeding with object layer initialization.

**Comment by krisis** [comment]
@harshavardhana Do you have any logs from minio server when this happens?

**Comment by harshavardhana** [comment]
Dont have any logs. But i suspect the issue is in xlHouseKeeping..

**Comment by harshavardhana** [comment]
The problem is that we are leaving behind a lot of 'tmp' files

```
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 521df168-ca5d-4152-a65c-e7280e62d70a
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 1cdfbb1f-4b34-4572-b584-01f9e9e547c4
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 ab9c6fb2-f5e3-4324-8839-5c4efdebfa9f
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 a7da30cb-18b5-49f0-bcd2-c12409aefc91
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 a02f1424-8954-423f-a578-1220ede1b596
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 f377afc2-a354-49d1-9c37-8e77dc195eb8
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 d88fcba2-7237-4876-ad12-83f853be4909
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 6674a7ff-b0e0-4623-8562-6cc21e74a89a
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 e1ee08b6-8c2d-4017-a40a-2c514eb90c2c
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 b36f4d88-72fd-42b7-819f-16690e404ae8
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 11a6a15f-5a8f-4f32-ba45-67899b42d4cd
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 920b3f7f-9387-4cc2-86ed-e4a710d0de8b
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 af0389f1-1aa4-45ee-a431-cc9e2af37374
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 e1865942-f13e-4ff3-9aea-284c759981b3
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 8717ab20-9d30-414e-ab9e-5285dcde59f7
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 347be065-99ae-499c-83e8-ade2442f7f95
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 581f8dff-c014-4250-814d-c760e620f4a2
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 e91b49a3-b421-4f7f-8a7d-aa62a5845574
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 7c9cf5c0-4ac4-4116-ba0a-6469ad71d1a1
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 f00eef47-4b12-47ac-8017-4226c5655cbe
drwxr-xr-x 2 minio s^C
drwxr-xr-x 2 minio sudo    4096 Oct 18 07:30 a1a1d2e6-6e0e-4461-9860-62d09e254518
minio@minio1:~$ ^C
minio@minio1:~$ ls -ltr /mnt/export{1,2}/minio/.minio.sys/tmp/*
```

**Comment by krisis** [comment]
`xlHouseKeeping` is an internal function that cleans up temporary directories that were created by API requests that didn't complete before the minio server instance was shutdown. It performs the clean up on all the disks. In a distributed Minio setup, each minio server exports its local disks via RPC. Evne before object layer is initialized the storage layer would be serving data to other minio servers participating. This opens up the possibility of the clean up activity interfering with legitimate temporary directories created by ongoing API requests like PutObject etc.
To address this, we should move "housekeeping" out of object layer initialization, as it stands today, and perform it before the storage layer starts serving data of the disks.

**Comment by harshavardhana** [comment]
Fixed by #3001
